pkg/io/serialize.go

```diff
@@ -62,6 +62,10 @@ func Encode(writer io.Writer, values ...interface{}) (err error) {
 				return errors.WithMessage(err, "writing length of marshalled data")
 			}
 
+			// Nothing to be encoded when length is zero.
+			if length == 0 {
```

Comment:
Couldn't it be that the serializer needs to write something to ensure that an object has been "registered"?

Reply:
Yes, we do write the length of data (which is zero). We just don't write the data itself. Because, there is nothing to write.